src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -89,10 +90,13 @@ def _enqueue_fs_event(self, event):
                       .format(len(event['files']), event['subscription']))
     self._event_queue.put(event)
 
-  def _maybe_invalidate_scheduler_batch(self, files):
-    invalidating_files = self._invalidating_files
-    if any(f in invalidating_files for f in files):
-      self._logger.fatal('saw file events covered by invalidation globs, terminating the daemon.')
+  def _maybe_invalidate_scheduler_batch(self):
+    new_snapshot = self._get_snapshot()
+    if new_snapshot is not None and self._invalidating_snapshot is not None and \
+      new_snapshot.directory_digest != self._invalidating_snapshot.directory_digest:
```

Comment:
I'm not sure what `self._get_snapshot()` will do if there are no globs, so I think that you should replace:
```
new_snapshot = self._get_snapshot()
if new_snapshot is not None and self._invalidating_snapshot is not None and \
  new_snapshot.directory_digest != self._invalidating_snapshot.directory_digest:
```
with
```
if self._invalidating_snapshot and \
  self._get_snapshot().directory_digest != self._invalidating_snapshot.directory_digest:
```

Reply:
Done